pkg/store/employee/employee.go

```diff
@@ -48,6 +69,94 @@ func (s *store) Search(filter SearchFilter, pagination model.Pagination) ([]*mod
 	return employees, total, db.Find(&employees).Error
 }
 
+func (s *store) Save(employee *model.Employee) error {
+	return s.db.Save(&employee).Error
+}
+
+func (s *store) UpdateGeneralInfo(body EditGeneralInfo, id string) (*model.Employee, error) {
+	employee := &model.Employee{}
+	// 1.2 update infor
+	employee.FullName = body.Fullname
+	employee.TeamEmail = body.Email
+	employee.PhoneNumber = body.Phone
+	employee.LineManagerID = body.LineManager
+	employee.DiscordID = body.DiscordID
+	employee.GithubID = body.GithubID
+	// 1.3 save to DB
+	return employee, s.db.Table("employees").Where("id = ?", id).Updates(&employee).Preload("Chapter").Preload("Seniority").Preload("Position").Preload("LineManager").Preload("EmployeeRoles").Preload("EmployeeRoles.Role").Find(&employee).Error
+}
+
+func (s *store) UpdateSkills(body EditSkills, id string) (*model.Employee, error) {
+
+	var employee *model.Employee
+
+	return employee, s.db.Transaction(func(tx *gorm.DB) error {
+		// get employee by employee id
+		err := tx.Where("id = ?", id).First(&employee).Error
+		if err != nil {
+			return err
+		}
+
+		// 1.1 delete all roles of the employee
+		employeeRole := model.EmployeeRoles{}
+		tx.Table("employee_roles").Where("employee_id = ?", id).Delete(&employeeRole)
```

Comment:
In the UI, we can see this is an employee position in the company. The account Roles table here has been used for the user permission feature.